examples/user_guide/Performance_and_Debugging.ipynb

```diff
@@ -73,7 +73,43 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Layouts\n",
+    "## Reuse sessions\n",
+    "\n",
+    "In order to render a page for a user visiting your application the Panel server creates a so called session. This session has to execute the code associated with your application and then render the page template to HTML. Once the frontend has rendered the initial template it will establish a Websocket connection and render the actual Panel components into the template. This means that in order for the user to see any actual content the entire application has to be executed. While this can be sped up through the use of caching it will always take a little while for the initial page to be rendered.\n",
+    "\n",
+    "Luckily, as long as certain conditions are met you can easily cache sessions ensuring that the initial page is rendered very quickly. You can enable this behavior by setting `pn.config.reuse_sessions = True` (or equivalently `pn.extension(..., reuse_sessions=True)`).\n",
+    "\n",
+    "So let us explore what conditions need to be met for it to be safe to enable this behavior. Luckily there really is only one condition:\n",
+    "\n",
+    "> The the placement and number of Panel components as well as any template configuration should not be dependent on any external state that can change.\n",
+    "\n",
+    "Now this of course needs some unpacking. First of all it's important to note that the actual contents or types of your Panel components doesn't matter. It is only about the overall configuration and placement, i.e.:\n",
+    "\n",
+    "- The number of items you have marked as `.servable` or added to a template's `main`, `sidebar` and `header` areas.\n",
+    "- Secondly, if you are using a `Template` its parameters should not change.\n",
+    "\n",
+    "Some examples include:\n",
+    "\n",
+    "- You may not load a dataset that might change and then add a variable number of plots.\n",
+    "- Set a template parameter based on some changing value `BootstrapTemplate(title=f'Dashboard at {dt.datetime.now()}')`\n",
+    "\n",
+    "One exception is when the template changes based on something in the HTTP request, e.g. the URL query parameters, headers or cookies. In these cases you have to tell Panel that there is some dependency on a request argument. Specifically we have to define a `pn.config.session_key_func`. Let's say for example that we switch the theme depending on a 'theme' query parameter. To ensure Panel caches the session for all possible theme values we define a function that returns a key that includes the entire URI `request.uri` (e.g. `/myapp?theme=dark`):\n",
```

Comment:
```suggestion
    "One exception is when the template changes based on something in the HTTP request, e.g. the URL query parameters, headers or cookies. In these cases you have to tell Panel that there is some dependency on a request argument. Specifically we have to define `pn.config.session_key_func`, giving it a callable that should return a key that will be used to cache the session. Let's say for example that we switch the theme depending on a 'theme' query parameter. To ensure Panel caches the session for all possible theme values we define a function that returns a key that includes the entire URI `request.uri` (e.g. `/myapp?theme=dark`):\n",
```

Reply:
Small suggestion to improve the docs based on your previous comment. I just missed that the returned value would be the cache key (even if it's written in the paragraph), so emphasizing that early.